subsys/bluetooth/host/smp.c

```diff
@@ -5211,6 +5244,52 @@ int bt_smp_auth_cb_overlay(struct bt_conn *conn, const struct bt_conn_auth_cb *c
 	}
 }
 
+#if defined(CONFIG_BT_PASSKEY_KEYPRESS_API)
+static int smp_send_keypress_notif(struct bt_smp *smp, uint8_t type)
+{
+	struct bt_smp_keypress_notif *req;
+	struct net_buf *buf;
+
+	buf = smp_create_pdu(smp, BT_SMP_KEYPRESS_NOTIFICATION, sizeof(*req));
+	if (!buf) {
+		return -ENOMEM;
+	}
+
+	req = net_buf_add(buf, sizeof(*req));
+	*req = (struct bt_smp_keypress_notif){
+		.type = type,
+	};
```

Comment:
Why not just
```suggestion
	req->type = type;
```

Reply:
I just think it makes the code easier to read. The struct type is visible right there. And it's in general a easier to read pattern because it's clear that you are overwriting all of the variable's contents, since any omitted fields are automatically zeroed out, so it contributes to code locality.

Reply:
I disagree with the readability :)

However, your point with automatically setting all other values to 0 is valid. It could be argued that in cases like this, it may be better to explicit set those to 0, so it's more clear that none were forgotten. 

I guess my main argument, besides disagreeing with your readability argument, is this is an uncommon pattern compared to the read of the BT subsystem, where we do this specific thing a lot of places. So even if it was more readable, then it is breaking an established pattern which on its own makes less readable IMO.